proc_macros/src/rule.rs

```diff
@@ -403,7 +420,9 @@ fn get_rule_rule_impl(
                 }
             }
 
-            fn instantiate(self: std::sync::Arc<Self>, _config: &#crate_name::Config) -> std::sync::Arc<dyn #crate_name::RuleInstance> {
+            fn instantiate(self: std::sync::Arc<Self>, _config: &#crate_name::Config, rule_configuration: &#crate_name::RuleConfiguration) -> std::sync::Arc<dyn #crate_name::RuleInstance> {
+                let options = rule_configuration.options.as_ref();
+                #maybe_deserialize_options
```

Comment:
I think the idea is that `options` will be "in scope" for the `per-run` state field initializers regardless, and if you don't provide an `options_type` then that `options` will be of the generic `RuleOptions` aka `serde_json::Value` type?